src/FormV2/UserCreation/CreateBeneficiaryType.php

```diff
@@ -34,22 +35,16 @@ public function __construct(
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
         $beneficiary = $options['data'];
-        $step = $options['step'];
-        $remotely = $beneficiary->getCreationProcess()?->isRemotely() ?? false;
-
-        if ($remotely) {
-            match ($step) {
-                default => $this->addStep1Fields($builder, $beneficiary, $remotely),
-                2 => $this->addStep4Fields($builder, $beneficiary),
-            };
-        } else {
-            match ($step) {
-                default => $this->addStep1Fields($builder, $beneficiary, $remotely),
-                2 => $this->addStep2Fields($builder),
-                3 => $this->addStep3Fields($builder, $beneficiary),
-                4 => $this->addStep4Fields($builder, $beneficiary),
-            };
-        }
+        $creationProcess = $beneficiary->getCreationProcess();
+        $step = $creationProcess?->getCurrentStep() ?? 1;
+        $remotely = $creationProcess?->isRemotely() ?? false;
+
+        match ($step) {
+            default => $this->addStep1Fields($builder, $beneficiary, $remotely),
+            2 => $this->addStep2Fields($builder, $beneficiary, $remotely),
```

Comment:
Peut-être plus simple de faire qqch comme `2 =>  $remotely ? $this->addStep4Fields($builder, $beneficiary) : $this->addStep2Fields($builder, $beneficiary);`